docs/src/languages/elixir.md

```diff
@@ -0,0 +1,165 @@
+```
+
+Include in your build hook the steps to retrieve a local Hex and rebar, and then run `mix do deps.get, deps.compile, compile` on your application to build a binary.
+
+{{< readFile file="src/registry/images/examples/full/elixir.hooks.app.yaml" highlight="yaml" >}}
+
+{{< note >}}
+The above build hook will work for most cases, and assumes that your `mix.exs` file is located at the root of your application.
+{{< /note >}}
+
+## Dependencies
+
+The recommended way to handle Elixir dependencies on Platform.sh is using Hex. You can commit a `mix.exs` file in your repository and the system will download the dependencies in your `deps` section using the build hook above.
+
+```elixir
+  defp deps do
+    [
+	  {:platformshconfig, "~> 0.1.0"}
+    ]
+  end
+```
+
+## Platform.sh variables
+
+Platform.sh exposes relationships and other configuration as [environment variables](/development/variables.md). Most notably, it allows a program to determine at runtime what HTTP port it should listen on and what the credentials are to access [other services](/configuration/services.md).
+
+To get the `PORT` environment variable (the port on which your web application is supposed to listen) you would:
+
+```elixir
+String.to_integer(System.get_env("PORT") || "8888")
+```
+
+Some of the environment variables are in JSON format and are base64 encoded. You would need to import a JSON parsing library such as [Jason](https://hexdocs.pm/json/readme.html) or [Poison](https://hexdocs.pm/poison/api-reference.html) to read those.
+
+{{< note title="Caveat">}}
+Remember `config/prod.exs` is evaluated at **build time** and will not have access to runtime configuration. Use `config/releases.exs` to configure your runtime environment.
+{{< /note >}}
+
+## Building and running the application
+
+Assuming `mix.exs` is present at the root of your repository and your build hook has been defined to match the [above example](#assumptions), you can then start it from the `web.commands.start` directive.  
+
+{{< note >}}
+The start command _must_ run in the foreground, so you should set the `--no-halt` flag when calling `mix run`.
+{{< /note >}}
+
+The following basic `.platform.app.yaml` file is sufficient to run most Elixir applications.
+
+```yaml
+name: app
+
+type: elixir:1.9
+
+variables:
+    env:
+        MIX_ENV: 'prod'
+        SECRET_KEY_BASE: $PLATFORM_PROJECT_ENTROPY
+
+hooks:
+    build: |
+        mix local.hex --force
+        mix local.rebar --force
+        mix do deps.get --only prod, deps.compile, compile
+
+web:
+    commands:
+        start: mix run --no-halt
+    locations:
+        /:
+            allow: false
+            passthru: true
+
+disk: 512
+```
+
+Note that there will still be a proxy server in front of your application.  If desired, certain paths may be served directly by our router without hitting your application (for static files, primarily) or you may route all requests to the Elixir application unconditionally, as in the example above.
```

Comment:
This isn't entirely correct.  The router doesn't serve it, the nginx proxy in the app container does.  How do we describe this on the Go, Node, or Java pages?

Reply:
@Crell Updated to match what's in Go for now:

> Note that there will still be an Nginx proxy server sitting in front of your application. If desired, certain paths may be served directly by Nginx without hitting your application (for static files, primarily) or you may route all requests to the Elixir application unconditionally, as in the example above.